### ISSUE TITLE
collaborations sync errors

### ISSUE THREAD

**Issue by nikolra** [excalidraw/excalidraw#6054]
When collaborating changes are not saved and are lost.

**Comment by dwelle** [comment]
Hi. Works for me. Can you reproduce consistently and provide steps?

**Comment by nikolra** [comment]
<img width="977" alt="Screenshot 2023-01-02 at 9 54 37" src="https://user-images.githubusercontent.com/81241995/210205809-acfc48dc-4a1f-432e-af8f-ada1017d26b7.png">
![WhatsApp Image 2023-01-02 at 09 54 24](https://user-images.githubusercontent.com/81241995/210205855-a9aff06e-9c31-4255-a1d9-0acd218333ba.jpeg)
There are screenshots from the same time of the same board from 2 different computers. The first picture is the more recent board. If I close the window on the first computer all the changes will be lost

**Comment by marcelatrujillo-takeblip** [comment]
Hey guys, I have the same problem here. I just lost everything I did in the board. Is it possible to recover that?

this what I had in the board:
![image](https://user-images.githubusercontent.com/121826120/210277076-d9fe6c87-1ece-49d6-9571-7869f7ac3724.png)

and today after a coworker have tried to access the same board, she wasn't seeing all the changes I did and in a couple minutes neither I saw it anymore. 

![image](https://user-images.githubusercontent.com/121826120/210277361-edeafed4-c4e4-4584-a073-c0f73b7c2951.png)

**Comment by dwelle** [comment]
From the images above it _may_ be that the canvas size exceeded the database limit of 1MB. For some stupid reason we're not showing an error message on these failures :(. Nonetheless, if a save fails, it shouldn't allow you to close the browser tab as it will prompt you that you may have unsaved data.

@nikolra @marcelatrujillo-takeblip Could you check your file sizes as of _now_? You can do so by right-clicking the canvas, opening "Stats for nerds" and checking the total storage size:

![image](https://user-images.githubusercontent.com/5153846/210283270-e0031971-daa8-4851-89d0-52184acaab36.png)

If it was a save failure, then there's no way to retrieve lost data, sorry :(. But even otherwise, it's likely the data weren't saved properly. The only case where we could retrieve the data is if the shapes were marked as deleted and the save happened within the last 24 hrs.

I've merged a fix to show the errors properly: https://github.com/excalidraw/excalidraw/pull/6063

**Comment by nikolra** [comment]
<img width="131" alt="Screenshot 2023-01-03 at 10 51 34" src="https://user-images.githubusercontent.com/81241995/210325323-46887637-c209-4ded-bef7-247dc9a787cc.png">
this is my stats

**Comment by dwelle** [comment]
@nikolra right, so size seems to be the issue indeed, as to the total sits just below 1MB.

**Comment by niokolra** [comment]
This was taken from the same canvas as the previous photos.

**Comment by marcelatrujillo-takeblip** [comment]
> From the images above it _may_ be that the canvas size exceeded the database limit of 1MB. For some stupid reason we're not showing an error message on these failures :(. Nonetheless, if a save fails, it shouldn't allow you to close the browser tab as it will prompt you that you may have unsaved data.
> 
> @nikolra @marcelatrujillo-takeblip Could you check your file sizes as of _now_? You can do so by right-clicking the canvas, opening "Stats for nerds" and checking the total storage size:
> 
> ![image](https://user-images.githubusercontent.com/5153846/210283270-e0031971-daa8-4851-89d0-52184acaab36.png)
> 
> If it was a save failure, then there's no way to retrieve lost data, sorry :(. But even otherwise, it's likely the data weren't saved properly. The only case where we could retrieve the data is if the shapes were marked as deleted and the save happened within the last 24 hrs.
> 
> I've merged a fix to show the errors properly: #6063

This is what appears to me :\

![image](https://user-images.githubusercontent.com/121826120/210359621-66047569-65f9-46ae-b4b1-392746cb8b0b.png)
